### PR TITLE
Replace jQuery UI Autocomplete for Tags with either HTML5 Datalist or Vanilla JS function

### PR DESCRIPTION
**Pull request by KTS915**
## Description

This PR follows the same objective as PR #1265, PR #1273, PR #1280, and PR #1284, and PR # 1286 in replacing functionality currently provided by jQuery UI with a native HTML5 implementation.

In this case, it is the jQuery UI Autocomplete module that is being replaced. In one file, `meta-boxes.php`, it is being replaced by the native HTML5 `datalist` element.

In the other file, `class-wp-posts-list-table.php`, the Autocomplete function is applied to `textarea`s (for each of Bulk and Quick Edit), and the `datalist` element does not work on `textarea`s (only on `input`s). For this reason, a new vanilla JS function has been added to `inline-post-edit.js` to maintain the same functionality (with a little new CSS to make it fit in).

As a result of these changes, the `tags-suggest.js` file will be deleted on grounds of redundancy.

## Motivation and context

The new code assists accessibility and reduces bloat. It also speeds up the relevant admin screens/pages in two ways.

First, on a post edit screen, it requires no JavaScript.

Secondly, whereas the current method makes a new call to the database each time a new character is added to the relevant input field, the new method makes such a call only once (on page load via PHP and not using JavaScript). Any keyboard interactions within the input field then trigger only a matching of strings within the browser; there are no further calls to the database. (This is true whether a `datalist` or the new vanilla JS function is used.)

## How has this been tested?

I have tested this on my localhost environment. (I'd hope it can be ready for CP v2.1, but it's certainly not intended for v2.0.)

## Screenshots

Appearance is unchanged; it's the underlying code that's different.

## Types of changes

New feature: this is not a breaking change since the jQuery UI modules continue to be bundled with ClassicPress and so remain available to any theme or plugin that's already relying on their presence; they are just not being used by core here. Indeed, once this and PR #1286 have been merged into core, core will not be using jQuery UI Autocomplete at all.

### CODE REVIEW

**Comment by xxsimoxx** [comment]
Before this PR, in both inline edit and edit post, autocomplete works for tag that are never been used.
Now it does not suggest tags created from the "tag" page but never used.

File: src/wp-admin/includes/meta-boxes.php [hunk 1]
```diff
@@ -583,10 +583,28 @@ function post_tags_meta_box( $post, $box ) {
 		<label for="tax-input-<?php echo $tax_name; ?>"><?php echo $taxonomy->labels->add_or_remove_items; ?></label>
 		<p><textarea name="<?php echo "tax_input[$tax_name]"; ?>" rows="3" cols="20" class="the-tags" id="tax-input-<?php echo $tax_name; ?>" <?php disabled( ! $user_can_assign_terms ); ?> aria-describedby="new-tag-<?php echo $tax_name; ?>-desc"><?php echo str_replace( ',', $comma . ' ', $terms_to_edit ); // textarea_escaped by esc_attr() ?></textarea></p>
 	</div>
-	<?php if ( $user_can_assign_terms ) : ?>
+
+	<?php
+	if ( $user_can_assign_terms ) :
+		$tags = get_tags(
+			array(
+				'fields' => 'names',
```
Comment (xxsimoxx) [inline src/wp-admin/includes/meta-boxes.php:591]: Adding this solve [my comment](https://github.com/ClassicPress/ClassicPress/pull/1287#issuecomment-1874038714).
`'hide_empty' => false,`

File: src/wp-admin/includes/class-wp-posts-list-table.php [hunk 2]
```diff
@@ -1841,11 +1841,24 @@ public function inline_edit() {
 					<?php foreach ( $flat_taxonomies as $taxonomy ) : ?>
 
 						<?php if ( current_user_can( $taxonomy->cap->assign_terms ) ) : ?>
-							<?php $taxonomy_name = esc_attr( $taxonomy->name ); ?>
+							<?php
+							$taxonomy_name = esc_attr( $taxonomy->name );
+
+							$tags = get_tags(
+								array(
+									'fields' => 'names',
```
Comment (xxsimoxx) [inline src/wp-admin/includes/class-wp-posts-list-table.php:1849]: Adding this solve [my comment](https://github.com/ClassicPress/ClassicPress/pull/1287#issuecomment-1874038714).
`'hide_empty' => false,`

File: src/wp-admin/includes/meta-boxes.php (hunk 1 above)
Comment (KTS915) [reply src/wp-admin/includes/meta-boxes.php]: Now added

File: src/wp-admin/includes/class-wp-posts-list-table.php (hunk 2 above)
Comment (KTS915) [reply src/wp-admin/includes/class-wp-posts-list-table.php]: Now added

**Review by xxsimoxx — APPROVED** [review]
LGTM

**Pull request merged**